migra/formatter.py

```diff
@@ -2,16 +2,79 @@
 
 class Formatter:
 
-    def __init__(self, eager_parens: bool=True) -> None:
+    def __init__(self, eager_parens=True, destructive_func_replace=True) -> None:
         # add other format wide configs
         self.eager_parens = eager_parens
+        self.destructive_func = destructive_func_replace
 
 
     def create_policy(self, all_changes):
         if self.eager_parens:
             # only fails on check true params, so hard coding is ok
             for i, change in enumerate(all_changes):
-                all_changes[i] = re.sub(r'with\s+check\s+true', 'with check (true)', change)
+                all_changes[i] = re.sub(r'with\s+check\s+true', 'with check (true)', change, flags=re.IGNORECASE)
         return all_changes
 
+    def prepend_drops_for_replacements(self, all_changes: list) -> list:
+        drops = []
+        if self.destructive_func:
+            for change in all_changes:
+                matches = re.search(r'CREATE\s+OR\s+REPLACE\s+FUNCTION\s+([^\)]+)', change, flags=re.IGNORECASE)
+                if matches:
+                    drop = f'\ndrop function if exists {matches.group(1)});\n'
+                    drops.append(drop)
+        return drops + all_changes
     
+if __name__ == '__main__':
```

Comment:
dont put tests here

put tests in a separate testing module in the root of the project and sits next to the library

Reply:
That's accurate, and what I do most of the time. Was getting some module import errors and didn't want to debug. Good crit